rustls/src/client/tls13.rs

```diff
@@ -479,9 +480,11 @@ impl hs::State for ExpectEncryptedExtensions {
                 let write_key = self
                     .key_schedule
                     .client_handshake_traffic_secret(
-                        &self
+                        self
                             .handshake
-                            .hash_at_client_recvd_server_hello,
+                            .hash_at_client_recvd_server_hello
+                            .as_ref()
+                            .unwrap(),
```

Comment:
I have a PR on top of this one to remove this `unwrap()`, by threading `hash_at_client_recvd_server_hello` through the handshake state machine directly.

Reply:
Why not just add the changes to this PR as a separate commit?

Reply:
I think that's out of scope for this PR. Note that if this `unwrap()` were to fail, then previously we were calculating the wrong value. And similarly for the other two `unwrap()`s I added in this PR.